### PR TITLE
Change enforce to sysctl

### PR DESCRIPTION
**Pull request by jaskarankhurana**
Change enforce from module parameter to sysctl so it can be changed at runtime without a reboot.

### CODE REVIEW

File: security/ipe/ipe.c [hunk 1]
```diff
@@ -3,10 +3,53 @@
...
 
-bool enforce;
+int enforce;
+
+#ifdef CONFIG_SYSCTL
+static int zero;
+static int one = 1;
+static struct ctl_table_header *ipe_sysctl_header;
+
+static const struct ctl_path ipe_sysctl_path[] = {
```
Comment (bowerscd) [inline security/ipe/ipe.c]: I assume this translates into `/ipe/policy`?

File: security/ipe/ipe.c [hunk 2]
```diff
@@ -3,10 +3,53 @@
...
 #include <linux/lsm_hooks.h>
+#include <linux/sysctl.h>
 #include "ipe.h"
 #include "ipe-hooks.h"
 
-bool enforce;
+int enforce;
+
+#ifdef CONFIG_SYSCTL
+static int zero;
```
Comment (bowerscd) [inline security/ipe/ipe.c]: the use of these two variables are not clear to me

File: security/ipe/ipe.c [hunk 3]
```diff
@@ -3,10 +3,53 @@
...
+static struct ctl_table_header *ipe_sysctl_header;
+
+static const struct ctl_path ipe_sysctl_path[] = {
+	{ .procname = "ipe", },
+	{ .procname = "policy", },
+	{ }
+};
+
+static struct ctl_table ipe_sysctl_table[] = {
+	{
```
Comment (bowerscd) [inline security/ipe/ipe.c]: 644 is fine - but confirming you want it to be 644.

I'm not sure what proc_dointvec_minmax is - does it just set the enforce number to .extra1 for min and .extra2 for max?

File: security/ipe/ipe.c (hunk 1 above)
Comment (jaskarankhurana) [reply security/ipe/ipe.c]: this would be ipe.policy.enforce=1 or 0 for enforce vs audit.

**Pull request merged**